Applications/Telemetry/telemetry.py

```diff
@@ -371,134 +416,206 @@ def stations():
+                data["BOARDTEMP"],
+                data["ADC8"],
+                data["HABTIMERSTATE"],
+                data["HABCUTDOWNSTATE"],
+                data["HABTRIGGERTIME"],
+                data["HABTIMER"],
+                data["EPOCH"]
+                ]
+
+    except KeyError as e:
+        logger.error("KeyError: " + str(e))
+        temp = []
 
     return temp
 
 
 def sqlInsert(data):
-    """Takes in a data tuple and inserts int into the telemetry SQLite table"""
+    """
+    Takes in a data tuple and inserts int into the telemetry SQLite table
 
-    # Read in name of telemetry databse
-    db = telemetryConfig.get("DATABASE", "FILENAME")
+    :param data: Telemetry dictionary
+    :return: Status True or False on SQL insertion success
+    """
+
+    # Read in name of telemetry database
+    try:
+        db = telemetryConfig.get("DATABASE", "FILENAME")
 
+    except ConfigParser.Error as e:
+        logger.error("ConfigParse.Error: " + str(e))
+        return False
+
+    # Change dictionary into list with proper order
     telem = createTelemetryList(data)
 
-    # Create parameter substitute "?" string for SQL query then create SQL
-    numKeys = len(telem)
-    paramSubs = "?" * (numKeys)
-    paramSubs = ",".join(paramSubs)
-    sql = "INSERT INTO TELEMETRY VALUES(" + paramSubs + ")"
+    # Check if length of telem is correct
+    if len(telem) > 0:
+        # Create parameter substitute "?" string for SQL query then create SQL
+        numKeys = len(telem)
+        paramSubs = "?" * (numKeys)
+        paramSubs = ",".join(paramSubs)
+        sql = "INSERT INTO TELEMETRY VALUES(" + paramSubs + ")"
+
+        # Connect to database, create SQL query, execute query, and close database
+        try:
+            conn = sqlite3.connect(db)
+
+        except sqlite3.Error as e:
+            logger.error("Sqlite3.Error: " + str(e))
+            return False
 
-    # Connect to database, create SQL query, execute query, and close database
-    try:
-        conn = sqlite3.connect(db)
         cursor = conn.cursor()
 
-        # Use connection as context manager to rollback automatically if error
-        with conn:
-            conn.execute(sql,telem)
+        try:
+            # Use connection as context manager to rollback automatically if error
+            with conn:
+                conn.execute(sql,telem)
 
-    except conn.ProgrammingError as e:
-        logger.error(e)
-        logger.error(telem)
-        conn.rollback()
-    except ValueError as e:
-        logger.error("ValueError: " + str(e))
-    except IndexError as e:
-        logger.error("IndexError: " + str(e))
-    except KeyError as e:
-        logger.error("KeyError: " + str(e))
-    except sqlite3.OperationalError as e:
-        # TODO: cleanup
-        logger.error(e)
-        initDB()
+        except sqlite3.Error as e:
+            logger.error("Sqlite3.Error: " + str(e))
+            conn.close()
+            return False
```

Comment:
You should do a roll-back to throw out SQL edits if error occurs to avoid corrupting database.